src/globus_cli/commands/api.py

```diff
@@ -263,102 +247,212 @@ def build_command(service_name: ServiceNameLiteral) -> click.Command:
-        for header_name, header_value in header:
-            headers_d[header_name] = header_value
-
-        # Strip `/v2` from Groups paths, which are auto-added by `GroupsClient`.
-        if service_name == "groups" and path.startswith("/v2"):
-            path = path[3:]
-
-        # try sending and handle any error
-        try:
-            res = client.request(
-                method.upper(),
-                path,
-                query_params=query_params_d,
-                data=body,
-                headers=headers_d,
+
+            client = _get_client(login_manager, service_name)
+            return _execute_service_command(
+                client,
+                method=method,
+                path=path,
+                query_param=query_param,
+                header=header,
+                body=body,
+                body_file=body_file,
+                content_type=content_type,
+                allow_errors=allow_errors,
                 allow_redirects=allow_redirects,
+                no_retry=no_retry,
+            )
+
+    else:
+
+        @command("gcs", help=helptext)
+        @LoginManager.requires_login("auth", "transfer")
+        @endpoint_id_arg
+        @_service_command_params
+        def service_command(
+            login_manager: LoginManager,
+            *,
+            endpoint_id: uuid.UUID,
+            method: Literal["HEAD", "GET", "PUT", "POST", "PATCH", "DELETE"],
+            path: str,
+            query_param: tuple[tuple[str, str], ...],
+            header: tuple[tuple[str, str], ...],
+            body: str | None,
+            body_file: t.TextIO | None,
+            content_type: Literal["json", "form", "text", "none", "auto"],
+            allow_errors: bool,
+            allow_redirects: bool,
+            no_retry: bool,
+            scope_string: tuple[str, ...],
+        ) -> None:
+            if scope_string:
+                _handle_scope_string(login_manager, str(endpoint_id), scope_string)
+
+            client = login_manager.get_gcs_client(endpoint_id=endpoint_id)
+            return _execute_service_command(
+                client,
+                method=method,
+                path=path,
+                query_param=query_param,
+                header=header,
+                body=body,
+                body_file=body_file,
+                content_type=content_type,
+                allow_errors=allow_errors,
+                allow_redirects=allow_redirects,
+                no_retry=no_retry,
             )
-        except globus_sdk.GlobusAPIError as e:
-            if not allow_errors:
-                raise
-            # we're in the allow-errors case, so print the HTTP response
-            print_error_or_response(e)
-        else:
-            print_error_or_response(res)
 
     return t.cast(click.Command, service_command)
 
 
 for service_name in CLI_SCOPE_REQUIREMENTS:
     api_command.add_command(build_command(service_name))
```

Comment:
Is this the troublemaker? I'd make this very much in-scope for this PR.

```suggestion
for service_name_ in CLI_SCOPE_REQUIREMENTS:
    api_command.add_command(build_command(service_name_))
```

Reply:
This is the troublemaker, indeed! We can make this change (I'll commit this suggestion after commenting), but I don't love that we still end up binding `service_name_` in module scope.

I was considering replacing this loop with the non-looping version with all of the names hardcoded in series. That's what I was considering out of scope for this PR.

Reply:
Oh! I think you can `del service_name_` after the loop. :+1:

Reply:
Oh, good call! Let me try after the merge.
That sounds best but I want this merged for other work to go smoothly.